Lib/test/test_json/test_decode.py

```diff
@@ -8,14 +8,34 @@
 class TestDecode:
     def test_decimal(self):
         rval = self.loads('1.1', parse_float=decimal.Decimal)
-        self.assertTrue(isinstance(rval, decimal.Decimal))
+        self.assertIsInstance(rval, decimal.Decimal)
         self.assertEqual(rval, decimal.Decimal('1.1'))
 
     def test_float(self):
         rval = self.loads('1', parse_int=float)
-        self.assertTrue(isinstance(rval, float))
+        self.assertIsInstance(rval, float)
         self.assertEqual(rval, 1.0)
 
+    def test_bytes(self):
+        self.assertEqual(self.loads(b"1"), 1)
+
+    def test_constant(self):
+        for constant, expected in [
+            ("Infinity", "INFINITY"),
+            ("-Infinity", "-INFINITY"),
+            ("NaN", "NAN"),
+        ]:
+            self.assertEqual(
+                self.loads(constant, parse_constant=str.upper), expected
```

Comment:
You can simply use `constant.upper()` instead of `expected`.

Reply:
I generally prefer to have explicit expected results and avoid computing them:

https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

But I can change it if you prefer.

Reply:
That's a good reason to do keep `expected`.

Reply:
To be consistent in avoiding computations in tests we should write this more explicitly:
```py
        self.assertEqual(self.loads("Infinity", parse_constant=str.upper),
                         "INFINITY")
        self.assertEqual(self.loads("-Infinity", parse_constant=str.upper),
                         "-INFINITY")
        self.assertEqual(self.loads("NaN", parse_constant=str.upper), "NAN")
```

If you want to allow some computations, you can write it as:
```python
        for name in "Infinity", "-Infinity", "NaN":
            self.assertEqual(self.loads(name, parse_constant=str.upper),
                             name.upper())
```
which is also more breaf and simple that the current code.

Reply:
I prefer the existing version:
* it shows the explicit inputs and outputs
* avoids extra computations
* we can immediately see we're asserting the function under test in the exact same way each time